src/posts/posts.service.spec.ts

```diff
@@ -1,18 +1,47 @@
 import { Test, TestingModule } from '@nestjs/testing';
+import { PostRepository } from './entities/post.repository';
```

Comment:
I couldn't find a place to mention this comment other than this one. please ensure that you test your repository implementation as it contain and will container more business logic

Reply:
I forgot to mention that you should keep the habit of logging things as you go. logging is a very important tool in your toolbox to maintain any application. using the combination of log levels and what to log where will enable you to easily debug production issues and analyze root causes.

we can discuss this whenever you are free.

Reply:
- Added tests for the repository implementation
- About logging: We already add logs for each request, and for each error. we will open anther PR when discussing any further requirements.